orangecontrib/network/widgets/OWNxTwitterGraph.py

```diff
@@ -0,0 +1,248 @@
+                                       callback=self.open_key_dialog,
+                                       tooltip="Set the API key for this widget.")
+        key_dialog_button.setFocusPolicy(QtCore.Qt.NoFocus)
+        box = gui.widgetBox(self.controlArea, "Info")
+        box.layout().addWidget(QtGui.QLabel("Users:"))
+        self.users = QtGui.QTextEdit()
+        box.layout().addWidget(self.users)
+        gui.label(box, self, 'Following: %(n_following)d\n'
+                             'Followers: %(n_followers)d\n'
+                             'All: %(n_all)d')
+        self.button = gui.button(box, self, "Create graph", self.fetch_users)
+
+    def open_key_dialog(self):
+        self.api_dlg.exec_()
+
+    def fetch_users(self):
+        CONSUMER_KEY = CredentialManager('Twitter API Key').key
+        CONSUMER_SECRET = CredentialManager('Twitter API Secret').key
+        OAUTH_TOKEN = ''
+        OAUTH_TOKEN_SECRET = ''
+        auth = twitter.oauth.OAuth(OAUTH_TOKEN, OAUTH_TOKEN_SECRET, CONSUMER_KEY, CONSUMER_SECRET)
+        t = twitter.Twitter(auth=auth)
+
+        followers_graph = nx.Graph()
+        following_graph = nx.Graph()
+        all_users = nx.Graph()
+        users = self.users.toPlainText().split("\n")
+        mapping = defaultdict(list)
+        fwers_id = 0
+        fwing_id = 0
+        all_id = 0
+
+        fwers_names=[]
+        fwing_names=[]
+        all_names=[]
+
+        for user in users:
+            result = t.users.show(screen_name=user)
+            id = result["id"]
+
+            if id in mapping.keys():
+                user_fwer_id, user_fwing_id, user_all_id = mapping[id]
+                if not user_fwer_id:
+                    mapping[id][0] = fwers_id
+                    user_fwer_id = fwers_id
+                    fwers_id += 1
+                    fwers_names.append([user, user])
+                else:
+                    fwers_names[user_fwer_id] = [user, user]
+                if not user_fwing_id:
+                    mapping[id][1] = fwing_id
+                    user_fwing_id = fwing_id
+                    fwing_id += 1
+                    fwing_names.append([user, user])
+                else:
+                    fwing_names[user_fwing_id] = [user, user]
+                all_names[user_all_id] = [user, user]
+            else:
+                mapping[id].extend([fwers_id, fwing_id, all_id])
+                user_fwer_id, user_fwing_id, user_all_id = fwers_id, fwing_id, all_id
+                fwers_id += 1
+                fwing_id += 1
+                all_id += 1
+                fwers_names.append([user, user])
+                fwing_names.append([user, user])
+                all_names.append([user, user])
+            followers_graph.add_node(user_fwer_id)
+            following_graph.add_node(user_fwing_id)
+            all_users.add_node(user_all_id)
+
+            cursor = -1
+            while cursor != 0:
```

Comment:
Same here; could the code from this and the following while be moved to a common method?